packages/react-native-reanimated/apple/reanimated/apple/REAAssertJavaScriptQueue.h

```diff
@@ -0,0 +1,14 @@
+#import <react/debug/react_native_assert.h>
+
+// Copied from RCTJSThreadManager.mm
+static NSString *const RCTJSThreadName = @"com.facebook.react.runtime.JavaScript";
+
+static BOOL REAIsJavaScriptQueue()
+{
+  return [NSThread.currentThread.name isEqualToString:RCTJSThreadName];
```

Comment:
Do we need to check it in Release mode too?

Reply:
The implementation of `RCTAssertJavaScriptQueue` is analogous to `RCTAssertMainQueue` (the only difference is that `RCTAssertMainQueue` is a precompiled macro and `RCTAssertJavaScriptQueue is a static inline function which is a bit more modern) so the behavior should be identical to the one in React Native. Besides, it seems like `RCTAssert` is optimized away in release build. Finally, these asserts are not expensive and as a matter of fact, having them also in the release build doesn't sound that bad at all.